src/Entity/MigrationProcessPipeline.php

```diff
@@ -0,0 +1,125 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_migration\Entity;
+
+use Drupal\Core\Config\Entity\ConfigEntityBase;
+
+/**
+ * Defines the Migration process pipeline entity.
+ *
+ * The migration process pipeline entity is a process plugin that can be
+ * exported to configuration and contains a suite of process plugins to run.
+ * It can be used in migrations as a single process plugin.
+ *
+ * @ConfigEntityType(
+ *   id = "oe_migration_process_pipeline",
+ *   label = @Translation("Migration Process Pipeline"),
+ *   module = "oe_migration",
+ *   handlers = {
+ *   },
+ *   entity_keys = {
+ *     "id" = "id",
+ *     "label" = "label"
+ *   },
+ *   config_export = {
+ *     "id",
+ *     "label",
+ *     "description",
+ *     "module",
+ *     "process",
+ *   },
+ * )
+ */
+class MigrationProcessPipeline extends ConfigEntityBase implements MigrationProcessPipelineInterface {
+
+  /**
+   * The migration process pipeline ID (machine name).
+   *
+   * @var string
+   */
+  protected $id;
+
+  /**
+   * The human-readable label for the migration process pipeline.
+   *
+   * @var string
+   */
+  protected $label;
+
+  /**
+   * The configuration describing the process plugins.
+   *
+   * @var array
+   */
+  protected $process = [];
```

Comment:
Seems this should be called `definitions` as they represent the definition of process plugins inside a migration.

Reply:
You're right, it's a better name. Renamed (and the related function  to get it was renamed too).